main.cpp

```diff
@@ -12,20 +12,28 @@
 #include <clang/Tooling/Tooling.h>
 #include <clang/Rewrite/Core/Rewriter.h>
 #include <clang/Frontend/CompilerInstance.h>
-
 using namespace clang;
 using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
+    CastCallBack(Rewriter& rewriter):r(rewriter) {
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto E = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");
+        auto b = E->getLParenLoc();
+        auto e = E->getRParenLoc();
+        auto est = E->getEndLoc();
+        r.RemoveText(b,1);
+        r.InsertText(b,"static_cast<");
+        r.RemoveText(e,1);
+        r.InsertText(e,">(");
+        r.InsertText(est.getLocWithOffset(1), ")");

```

Comment:
This code does not work as expected for the following example:
```cpp
// test.cpp
int main() {
    float fooooo;
    int i = (int)   fooooo   ;
    return 0;
}
```

Reply:
solved in new commit